cmd/contour/serve.go

```diff
@@ -482,8 +482,13 @@ func (s *Server) doServe() error {
 		}),
 	}
 
-	// snapshotHandler is used to produce new snapshots when the internal state changes for any xDS resource.
-	snapshotHandler := xdscache.NewSnapshotHandler(resources, s.log.WithField("context", "snapshotHandler"))
+	// snapshotHandler triggers go-control-plane Snapshots based on
+	// the contents of the Contour xDS caches after the DAG is built.
+	snapshotHandler := xdscache_v3.NewSnapshotHandler(
+		resources,
+		envoy_cache_v3.NewSnapshotCache(false, &contour_xds_v3.Hash, s.log.WithField("context", "snapshotCache")),
```

Comment:
This is the important change, the snapshot handler is now connected to the snapshot cache before any runnables are started.